probeservices/client_test.go

```diff
@@ -204,3 +206,385 @@ func TestSortEndpoints(t *testing.T) {
 		t.Fatal(diff)
 	}
 }
+
+func TestOnlyHTTPS(t *testing.T) {
+	in := []model.Service{{
+		Type:    "onion",
+		Address: "httpo://jehhrikjjqrlpufu.onion",
+	}, {
+		Type:    "https",
+		Address: "https://ams-ps.ooni.io",
+	}, {
+		Type:    "https",
+		Address: "https://hkg-ps.ooni.io",
+	}, {
+		Front:   "dkyhjv0wpi2dk.cloudfront.net",
+		Type:    "cloudfront",
+		Address: "https://dkyhjv0wpi2dk.cloudfront.net",
+	}, {
+		Type:    "https",
+		Address: "https://mia-ps.ooni.io",
+	}}
+	expect := []model.Service{{
+		Type:    "https",
+		Address: "https://ams-ps.ooni.io",
+	}, {
+		Type:    "https",
+		Address: "https://hkg-ps.ooni.io",
+	}, {
+		Type:    "https",
+		Address: "https://mia-ps.ooni.io",
+	}}
+	out := probeservices.OnlyHTTPS(in)
+	diff := cmp.Diff(out, expect)
+	if diff != "" {
+		t.Fatal(diff)
+	}
+}
+
+func TestOnlyFallbacks(t *testing.T) {
+	// put onion first so we also verify that we sort the endpoints
+	in := []model.Service{{
+		Type:    "onion",
+		Address: "httpo://jehhrikjjqrlpufu.onion",
+	}, {
+		Type:    "https",
+		Address: "https://ams-ps.ooni.io",
+	}, {
+		Type:    "https",
+		Address: "https://hkg-ps.ooni.io",
```

Comment:
I would suggest replacing these with another string that is clearly not used, as these domain names are very similar to one that exists, but in fact they do not.

Reply:
https://github.com/ooni/probe-engine/issues/664